### DISCUSSION TITLE
Fixing compatibility with plugin CreeperHeal2

### DISCUSSION THREAD

**Discussion by pmdevita** [TownyAdvanced/Towny#7283] [General]
Hey, I'm the author of [CreeperHeal2](https://github.com/pmdevita/CreeperHeal2). I'm trying to fix compatibility between my plugin and Towny and EventWar for one of my users and I'm running into some trouble.

It looks like EventWar is duplicating blocks in an explosion event's (`EntityExplodeEvent`, `BlockExplodeEvent`) `blockList` for the blocks inside a warzone. CreeperHeal2 watches this event on HIGH priority and I'm seeing duplicate entries for each block in the explosion. This seems to be happening even if EventWar's regeneration is turned off.

Is this intentional? It seems like it could be a bug that could easily slip past so I just want to make sure.

If you're handling your own regeneration, it might also be best to mask out warzones to CreeperHeal. Is there a maven repo for EventWar that I could use to create the dependency in CreeperHeal? We could also add the event listener to EventWar instead if you'd prefer not to open the API.

Thank you!

**Comment by LlmDl** [comment]
EventWar uses Towny's API to listen for blocks that are exploded, using the Towny TownBlockExplosionTestEvent event, which is kicked off from a few places which Towny listens to on HIGH priority. 

Towny and EventWar do handle their on regenerations. If you wanted to you can not heal explosions that are happening where towns have an active war, town.hasActiveWar() is the way to do it. Drop into the Discord and we can talk it over. I'm going afk for the evening.
